src/routes/biotopes/aquariums.ts

```diff
@@ -0,0 +1,262 @@
+                            200: AquariumDtoSchema,
+                        },
+                    },
+                },
+                async function (req, res) {
+                    const biotope = req.biotope!;
+
+                    biotope.archivedDate = new Date();
+                    await biotope.save();
+
+                    res.send(
+                        AquariumDtoSchema.parse({
+                            ...biotope.dataValues,
+                            ...biotope.AquariumModel!.dataValues,
+                        }),
+                    );
+                },
+            );
+
+            instance.put(
+                "/unarchive",
+                {
+                    schema: {
+                        tags: ["aquariums"],
+                        description: "Unarchive an aquarium",
+                        params: z.object({
+                            id: z.string().uuid(),
+                        }),
+                        response: {
+                            200: AquariumDtoSchema,
+                        },
+                    },
+                },
+                async function (req, res) {
+                    const biotope = req.biotope!;
+
+                    biotope.archivedDate = null;
+                    await biotope.save();
+
+                    res.send(
+                        AquariumDtoSchema.parse({
+                            ...biotope.dataValues,
+                            ...biotope.AquariumModel!.dataValues,
+                        }),
+                    );
+                },
+            );
+
+            /**
+             * @deprecated use /:id/measurements/TEMPERATURE instead
+             * Concerve for aquatracker v1 compatibility
+             */
+            instance.post(
+                "/temperature",
+                {
+                    schema: {
+                        tags: ["aquariums"],
+                        description: "Add temperature measurement of aquarium",
+                        deprecated: true,
+                        params: z.object({
+                            id: z.string().uuid(),
+                        }),
+                        body: z.object({
+                            temperature: z.number(),
+                        }),
+                        response: {
+                            200: z.null(),
+                        },
+                    },
+                },
+                async function (req, res) {
+                    const biotope = req.biotope!;
+
+                    await biotope.createMeasurementModel({
+                        measurementTypeCode: "TEMPERATURE",
+                        value: req.body.temperature,
+                    });
+
+                    res.send();
+                },
+            );
+
+            await fastify.register(import("./generics"));
+        },
+        {
+            prefix: "/:id",
+        },
+    );
+}) satisfies FastifyPluginAsync;
```

Comment:
Note that this could be a `FastifyPluginAsyncZod` instead. If I understood correctly, this allows you to remove this line because fastify is already correctly typed.
```ts
const instance = fastify.withTypeProvider<ZodTypeProvider>();
```

Reply:
I don't find `FastifyPluginAsyncZod` type

Reply:
There is a commit that adds this type to the lib, but the maintainers haven't updated the lib for 1 year on NPM. 🥲